pkg/gofr/datasource/file/s3/file.go

```diff
@@ -0,0 +1,475 @@
+//
+// Note:
+//   - This method should be called on a FileInfo instance obtained from a Stat or ReadDir operation.
+func (f *s3file) Size() int64 {
+	bucketName := strings.Split(f.name, string(filepath.Separator))[0]
+
+	f.sendOperationStats(&FileLog{
+		Operation: "FILE/DIR SIZE",
+		Location:  getLocation(bucketName),
+	}, time.Now())
+
+	return f.size
+}
+
+// ModTime returns the last modification time of the file or directory.
+//
+// For files, it returns the timestamp of the last modification to the file's contents.
+// For directories, it returns the timestamp of the most recent change to the directory's contents, including updates
+// to files within the directory.
+func (f *s3file) ModTime() time.Time {
+	bucketName := strings.Split(f.name, string(filepath.Separator))[0]
+
+	f.sendOperationStats(&FileLog{
+		Operation: "LAST MODIFIED",
+		Location:  getLocation(bucketName),
+	}, time.Now())
+
+	return f.lastModified
+}
+
+// IsDir checks if the FileInfo describes a directory.
+//
+// This method returns true if the FileInfo object represents a directory; otherwise, it returns false.
+// It is specifically used to determine the type of the file system object represented by the FileInfo.
+//
+// Note:
+//   - This method should be called on a FileInfo instance obtained from a Stat or ReadDir operation.
+//   - The FileInfo interface is used to describe file system objects, and IsDir is one of its methods
+//     to query whether the object is a directory.
+func (f *s3file) IsDir() bool {
+	bucketName := strings.Split(f.name, string(filepath.Separator))[0]
+
+	f.sendOperationStats(&FileLog{
+		Operation: "IS DIR",
+		Location:  getLocation(bucketName),
+	}, time.Now())
+
+	return f.name[len(f.name)-1] == '/'
+}
+
+// Close closes the response body returned in Open/Create methods if the response body is not nil.
+func (f *s3file) Close() error {
+	bucketName := strings.Split(f.name, string(filepath.Separator))[0]
+
+	defer f.sendOperationStats(&FileLog{
+		Operation: "CLOSE",
+		Location:  getLocation(bucketName)}, time.Now())
+
+	if f.body != nil {
+		return f.body.Close()
+	}
+
+	return nil
+}
+
+// Read reads data into the provided byte slice.
+//
+// It attempts to fill the entire slice with data from the file. If the number of bytes read is less than the length of the slice,
+// it returns an io.EOF error to indicate that the end of the file has been reached before the slice was completely filled.
+//
+// Additionally, this method updates the file offset to reflect the next position that will be used for subsequent read or write operations.
+func (f *s3file) Read(p []byte) (n int, err error) {
+	var fileName, msg string
+
+	bucketName := strings.Split(f.name, string(filepath.Separator))[0]
+
+	// get path relative to current bucketName
+	index := strings.Index(f.name, string(filepath.Separator))
+	if index != -1 {
+		fileName = f.name[index+1:]
+	}
```

Comment:
Please look at filepath package

Reply:
I don't think file path package has any function like this.